spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java

```diff
@@ -92,18 +95,26 @@
 	@Value("${spring.cloud.task.closecontext.enable:true}")
 	private Boolean closeContext;
 
+	@Value("${spring.cloud.task.executionid:}")
+	private Integer taskExecutionId;
+
+	@Value("${spring.cloud.task.external.executionid:#{null}}")
```

Comment:
This really should be `spring.cloud.task.external-execution-id` (no extra `.` and use `-` to separate multi word like boot).